test/components/selectionBoxLayerTests.ts

```diff
@@ -144,30 +162,101 @@ describe("Interactive Components", () => {
+        xScale.domain([0, 2000]);
+        xScale.range([0, svgWidth]);
+        sbl.xScale(xScale);
+
+        let bounds = {
+          topLeft: {
+            x: 100,
+            y: 100
+          },
+          bottomRight: {
+            x: 200,
+            y: 200
+          }
         };
-        sbl.bounds({
-          topLeft: topLeft,
-          bottomRight: bottomRight
-        });
+        sbl.bounds(bounds);
+        assert.deepEqual(sbl.bounds(), bounds, "latest bounds set");
+        let xExtent = [100, 250];
+        assert.strictEqual(sbl.xExtent(xExtent), sbl, "returns calling object");
 
-        assert.strictEqual(sbl.xExtent()[0], xScale.invert(100), "left data value maps correctly");
-        assert.strictEqual(sbl.xExtent()[1], xScale.invert(250), "right data value maps correctly");
+        assert.deepEqual(sbl.xExtent(), xExtent, "xExtent set");
+        assert.strictEqual(sbl.bounds().topLeft.x, xScale.scale(xExtent[0]), "left pixel position adjusts accordingly");
+        assert.strictEqual(sbl.bounds().bottomRight.x, xScale.scale(xExtent[1]), "right pixel position adjusts accordingly");
+
+        sbl.destroy();
+        svg.remove();
+      });
+
+      it("updates left and right edge pixel positions if in VALUE mode and scale is switched", () => {
+        let xScale = new Plottable.Scales.Linear();
+        xScale.domain([0, 2000]);
+        xScale.range([0, svgHeight]);
+        sbl.xScale(xScale);
+
+        let xExtent = [100, 250];
+        sbl.xExtent(xExtent);
+        assert.strictEqual(sbl.bounds().topLeft.x, xScale.scale(xExtent[0]), "left pixel position adjusts accordingly");
+        assert.strictEqual(sbl.bounds().bottomRight.x, xScale.scale(xExtent[1]), "right pixel position adjusts accordingly");
+
+        let xScale2 = new Plottable.Scales.Linear();
+        xScale2.domain([0, 1000]);
+        xScale2.range([0, svgHeight]);
+
+        sbl.xScale(xScale2);
+        assert.strictEqual(sbl.bounds().topLeft.x, xScale2.scale(xExtent[0]), "left pixel position adjusts accordingly");
+        assert.strictEqual(sbl.bounds().bottomRight.x, xScale2.scale(xExtent[1]), "right pixel position adjusts accordingly");
+
+        sbl.destroy();
+        svg.remove();
+      });
+
+      it("updates left and right edge pixel positions if in VALUE mode and scale updates", () => {
+        let xScale = new Plottable.Scales.Linear();
+        xScale.domain([0, 2000]);
+        xScale.range([0, svgHeight]);
+        sbl.xScale(xScale);
+
+        let xExtent = [100, 250];
+        sbl.xExtent(xExtent);
+        assert.strictEqual(sbl.bounds().topLeft.x, xScale.scale(xExtent[0]), "left pixel position adjusts accordingly");
+        assert.strictEqual(sbl.bounds().bottomRight.x, xScale.scale(xExtent[1]), "right pixel position adjusts accordingly");
+
+        xScale.domain([0, 1000]);
+
+        assert.strictEqual(sbl.bounds().topLeft.x, xScale.scale(xExtent[0]), "left pixel position adjusts accordingly");
```

Comment:
I don't really like these blind tests. Can you also assert that the xExtent[0] now is different than xExtent[0] before the scale update?